fdbserver/storageserver.actor.cpp

```diff
@@ -4864,9 +4875,22 @@ ACTOR Future<GetMappedKeyValuesReply> mapKeyValues(StorageServer* data,
 			                      "storageserver.mapKeyValues.AfterBatch");
 		subqueries.clear();
 		for (int i = 0; i + offset < sz && i < SERVER_KNOBS->MAX_PARALLEL_QUICK_GET_VALUE; i++) {
+			// since we always read the index, so always consider the index size
+			int indexSize = sizeof(KeyValueRef) + input.data[i + offset].expectedSize();
+			int size = indexSize + getMappedKeyValueSize(kvms[i]);
+			*remainingLimitBytes -= size;
+			++cnt;
 			result.data.push_back(result.arena, kvms[i]);
+			if (SERVER_KNOBS->STRICTLY_ENFORCE_BYTE_LIMIT && *remainingLimitBytes <= 0) {
+				// store index for this ending boundary, if it is already an ending boundary, no-op
+				result.data.back().key = input.data[i + offset].key;
+				result.data.back().value = input.data[i + offset].value;
```

Comment:
Do we similarly need to set the results `back()` key and value if we terminate early because of the byte limit after a batch (that is, if the `STRICTLY_ENFORCE_BYTE_LIMIT` knob is false)?

Reply:
in `mapSubQuery(`, we set the KV if `isBoundary` is true. I agree it is a bit complicated here -- we need to set both the [0] and [indexResultSize - 1], so we have `bool isBoundary = (i + offset) == 0 || (i + offset) == sz - 1;` and pass it  to `mapSubQuery(`.

So this code is only for "early" termination, KV will be set in `mapSubQuery(` in normal termination(i.e. return all entries)

Reply:
I misunderstood this comment, this is a good catch, I missed it, will fix.